cloudconfig.go

```diff
@@ -0,0 +1,71 @@
+package cloudconfig
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"html/template"
+	"io/ioutil"
+
+	"github.com/giantswarm/awstpr"
+	tpraws "github.com/giantswarm/awstpr/aws"
+	"github.com/giantswarm/clustertpr/node"
+)
+
+type AWSNode struct {
+	Node    node.Node
+	AwsInfo tpraws.Node
+}
+
+type CloudConfigTemplateParams struct {
+	Spec      awstpr.Spec
+	Node      AWSNode
+	TLSAssets CompactTLSAssets
+}
+
+type CloudConfig struct {
+	config   string
+	params   CloudConfigTemplateParams
+	path     string
+	template string
+}
+
+func NewCloudConfig(templatePath string, params CloudConfigTemplateParams) (*CloudConfig, error) {
+	cloudConfigTempl, err := ioutil.ReadFile(templatePath)
```

Comment:
I think the point was for example to have it get a `io.Reader` so you can use mock cloudconfigs and run unit tests against the package. Maybe afero is a bit overkill for this.